### PR TITLE
Add FB info

### PR DESCRIPTION
**Pull request by emersion**
Closes: https://github.com/ascent12/drm_info/issues/45

### CODE REVIEW

File: json.c [hunk 1]
```diff
@@ -290,6 +290,29 @@ static struct json_object *path_info(int fd, uint32_t blob_id)
...
 }
 
+static struct json_object *fb_info(int fd, uint32_t id)
+{
+	if (id == 0) {
+		return NULL;
+	}
+
+	// TODO: use drmModeGetFB2: https://patchwork.freedesktop.org/series/67552/
+	drmModeFB *fb = drmModeGetFB(fd, id);
```
Comment (ascent12) [inline json.c]: This seems to fail with EINVAL for me when used on the primary plane's FB, but the cursor one worked fine.
I haven't checked the kernel source, but maybe it doesn't like this when modifiers are in use, or something like that.

This call should check the return value.

File: json.c [hunk 2]
```diff
@@ -290,6 +290,29 @@ static struct json_object *path_info(int fd, uint32_t blob_id)
 	return obj;
 }
 
+static struct json_object *fb_info(int fd, uint32_t id)
+{
+	if (id == 0) {
+		return NULL;
+	}
```
Comment (ascent12) [inline json.c]: id is already checked before calling this function. Remove either this or the check before this function is called.

**Comment by emersion** [comment]
Done!

**Comment by emersion** [comment]
Meson's wrapdb seems down without a clear fix:

```
12:55 <woodwose> g'morn, how comes https://wrapdb.mesonbuild.com/ is down so often?
17:57 <xclaesse> woodwose: the host is down, it needs new hosting and possibly mirrors
```

**Pull request merged**